platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -283,581 +158,373 @@ namespace platoon_strategic
-                double maxJoinGap = std::max(config_.desiredJoinGap, desiredJoinGap2);
-                double currentGap = pm_.getDistanceToFrontVehicle();
-                ROS_DEBUG_STREAM("Based on desired join time gap, the desired join distance gap is " << desiredJoinGap2 << " ms");
-                ROS_DEBUG_STREAM("Since we have max allowed gap as " << desiredJoinGap << " m then max join gap became " << maxJoinGap << " m");
-                ROS_DEBUG_STREAM("The current gap from radar is " << currentGap << " m");
-                // TODO: temporary
-                if(true)//(currentGap <= maxJoinGap && pm_.current_plan.valid == false) {
-                {
-                    cav_msgs::MobilityRequest request;
-                    std::string planId = boost::uuids::to_string(boost::uuids::random_generator()());
-                    long currentTime = ros::Time::now().toNSec()/1000000; 
-                    request.header.plan_id = planId;
-                    request.header.recipient_id = pm_.targetLeaderId;
-                    request.header.sender_bsm_id = host_bsm_id_;
-                    request.header.sender_id = config_.vehicleID;
-                    request.header.timestamp = currentTime;
-                    request.plan_type.type = cav_msgs::PlanType::PLATOON_FOLLOWER_JOIN;
-                    request.strategy = MOBILITY_STRATEGY;
-                    request.strategy_params = "";
-                    request.urgency = 50;
-                    request.location = pose_to_ecef(pose_msg_);
-                    mobility_request_publisher_(request);
-                    ROS_DEBUG_STREAM("Published Mobility Candidate-Join request to the leader");
-                    ROS_WARN("Published Mobility Candidate-Join request to the leader");
-                    PlatoonPlan* new_plan = new PlatoonPlan(true, currentTime, planId, pm_.targetLeaderId);
-
-                    pm_.current_plan = *new_plan;
-                }
-        
-         //Task 4
-                if(pm_.getTotalPlatooningSize() > 1) {
-                    cav_msgs::MobilityOperation status;
-                    status = composeMobilityOperationCandidateFollower();
-                    mobility_operation_publisher_(status);
-                    ROS_DEBUG_STREAM("Published platoon STATUS operation message");
-                }
-                long tsEnd =  ros::Time::now().toNSec()/1000000; 
-                long sleepDuration = std::max((int32_t)(statusMessageInterval_ - (tsEnd - tsStart)), 0);
-                ros::Duration(sleepDuration/1000).sleep();
-        
-    }
+    //-------------------------------- Mobility Communication --------------------------------------//
 
-    MobilityRequestResponse PlatoonStrategicPlugin::handle_mob_req(const cav_msgs::MobilityRequest& msg)
+    // ------ 1. compose messages ------
+    cav_msgs::PlatooningInfo PlatoonStrategicIHPPlugin::composePlatoonInfoMsg()
     {
```

Comment:
The state definitions in this code (STANDBY, LEADER, LEADERWAITING, etc), are being mapped to the state definitions in the cav_msgs::PlatooningInfo (DISABLED, SEARCHING, LEADING, etc).  Ideally, these two sets of states would be identical.  I supposed it would be a huge amount of work at this point to refactor to make them identical.  Therefore, it would be useful to have a comment block (probably in the .h file) explaining a) why they are different, and b) how exactly the two sets map to each other.  Just as important, the algorithm doc Fig 2 should be updated to reflect the code's actual state model.  Or there needs to be a careful explanation for the differences in the code comment block.

Reply:
Indeed. There is a disconnection between "platooning Info status" versus "platoon strategic state". The "platoon info status" is a general status that reflect the ultimate goal. Whereas the "platoon strategic state" is more refined and is closely related to negotiation rules. For example "leader aborting" and "candidate follower" can all be mapped to "connecting to new leader" because both states are all serving the purpose of connecting to a new leader. 

I will add a note in both .h file demonstrating and at the top of the function (breifly) explaining the difference. The changes will be in next commit. And for the design doc, I will update the diagram.

Reply:
Sounds good.  Thanks!